rockcraft/application.py

```diff
@@ -37,7 +40,7 @@ class Rockcraft(Application):
 
     @override
     def _extra_yaml_transform(self, yaml_data: dict[str, Any]) -> dict[str, Any]:
-        return models.transform_yaml(self._work_dir, yaml_data)
+        return models.transform_yaml(Path.cwd(), yaml_data)
```

Comment:
I'm not sure I understand this change. I don't oppose it, I'm just curious what it fixes.

Reply:
I should have documented this better, but the parameter that should be passed to `models.transform_yaml()` is the directory that contains the project file. In non-managed mode `self._work_dir` is the cwd but in managed mode [it isn't](https://github.com/canonical/craft-application/blob/main/craft_application/application.py#L143).
I think `self.project_dir` didn't exist yet when I made this change, we should use that instead.